File: tests/fixtures/negative/android-lint/WebViewDebuggingEnabled.java
package test;

import android.content.pm.ApplicationInfo;
import android.webkit.WebView;

class App {
    void guardedByBuildConfig() {
        if (BuildConfig.DEBUG) {
            WebView.setWebContentsDebuggingEnabled(true);
        }
    }

    void guardedByApplicationInfo(ApplicationInfo applicationInfo) {
        if ((applicationInfo.flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0) {
            WebView.setWebContentsDebuggingEnabled(true);
        }
    }

    void argumentCarriesGuard() {
        WebView.setWebContentsDebuggingEnabled(BuildConfig.DEBUG);
    }
}

File: tests/fixtures/negative/android-lint/WebViewDebuggingEnabled.kt
package test

import android.content.pm.ApplicationInfo
import android.webkit.WebView

class App {
    fun guardedByBuildConfig() {
        if (BuildConfig.DEBUG) {
            WebView.setWebContentsDebuggingEnabled(true)
        }
    }

    fun guardedByApplicationInfo(applicationInfo: ApplicationInfo) {
        if ((applicationInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE) != 0) {
            WebView.setWebContentsDebuggingEnabled(true)
        }
    }

    fun argumentCarriesGuard() {
        WebView.setWebContentsDebuggingEnabled(BuildConfig.DEBUG)
    }
}
